ports/stm32f4/common-hal/busio/I2C.c

```diff
@@ -32,52 +32,129 @@
 
 #include "shared-bindings/microcontroller/__init__.h"
 #include "supervisor/shared/translate.h"
+#include "common-hal/microcontroller/Pin.h"
 
-I2C_HandleTypeDef hi2c2;
+void i2c_reset(void) {
+    //TODO: implement something better than eratta workaround. 
```

Comment:
You likely need a bulk reset here once you actually track pin usage.

Reply:
Again, I'm not sure what you mean by this, as pin resets are handled by reset_all_pins.

Reply:
True again. Maybe just reset any in use I2C peripherals then.

Reply:
The reason I2C peripherals are not reset here is because it's brute forced in the init code, as noted! I didn't feel like there was much point in doing it twice. If/when I do the peripheral rework I can either fill this with useful stuff or just delete it. Only nrf also has a function like this, I think.

Reply:
You'll want to do it on reset if you use the peripheral state to determine if it's already in use like samd does. I think it's still good to reset because it'll stop anything from happening after Python code finishes. There can be a big gap between python vm finishing and then rerunning again and the hardware should be in a known state during that time.